src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java

```diff
@@ -39,30 +47,54 @@ public void canRenderTemplates(TemplateRenderingTestCase testCase) throws Except
 
         // Event Errors (this is a template for an email that is sent in BugsnagEventHandlingJob#sendEmailForEvents)
         Map<String, String> errorEventsData = Map.of("subject", "test subject");
-        testCases.add(new TemplateRenderingTestCase(errorEventsData, "EventErrorsText.ftl"));
-        testCases.add(new TemplateRenderingTestCase(errorEventsData, "EventErrorsHtml.ftl"));
+        testCases.add(new TemplateRenderingTestCase(
+            errorEventsData, "EventErrorsText.ftl", "Event Errors Text Email"
+        ));
+        testCases.add(new TemplateRenderingTestCase(
+            errorEventsData, "EventErrorsHtml.ftl", "Event Errors Html Email"
+        ));
+
         // Trip Monitor Notifications tests (for CheckMonitoredTrip#sendNotifications).
         Map<String, Object> notificationsData = Map.of(
             "tripId", "18f642d5-f7a8-475a-9469-800129e6c0b3",
             "notifications", List.of("Test notification.", "Another notification.")
         );
-        testCases.add(new TemplateRenderingTestCase(notificationsData, "MonitoredTripSms.ftl"));
-        testCases.add(new TemplateRenderingTestCase(notificationsData, "MonitoredTripText.ftl"));
-        testCases.add(new TemplateRenderingTestCase(notificationsData, "MonitoredTripHtml.ftl"));
+        testCases.add(new TemplateRenderingTestCase(
+            notificationsData, "MonitoredTripSms.ftl", "Monitored Trip SMS"
+        ));
+        testCases.add(new TemplateRenderingTestCase(
+            notificationsData, "MonitoredTripText.ftl", "Monitored Trip Text Email"
+        ));
+        testCases.add(new TemplateRenderingTestCase(
+            notificationsData, "MonitoredTripHtml.ftl", "Monitored Trip HTML Email"
+        ));
         return testCases;
     }
 
     private static class TemplateRenderingTestCase {
+        private static final Set<String> testCaseNames = new HashSet<>();
         public final Object data;
         public final String templatePath;
+        public final String testCaseName;
 
-        private TemplateRenderingTestCase(Object data, String templatePath) {
+        private TemplateRenderingTestCase(Object data, String templatePath, String testCaseName) {
+            // enforce unique test case names so that snapshotting works properly
+            if (testCaseNames.contains(testCaseName)) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        "A test case with the name %s already exists! Test case names must be unique.",
+                        testCaseName
+                    )
+                );
+            }
+            testCaseNames.add(testCaseName);
```

Comment:
Non-blocking nit: we could just check `if (!testCaseNames.add(testCaseName))` above rather than a contains and add call. I don't feel strongly about this though (one could argue that the current way is more readable).